dotnet/src/SemanticKernel.Skills/Skills.Grpc/GrpcOperationRunner.cs

```diff
@@ -0,0 +1,253 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Grpc.Net.Client;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Skills.Grpc.Model;
+using ProtoBuf;
+using Skills.Grpc;
+
+namespace Microsoft.SemanticKernel.Skills.Grpc;
+
+/// <summary>
+/// Runs gRPC operation runner.
+/// </summary>
+internal class GrpcOperationRunner : IGrpcOperationRunner
+{
+    /// <summary>
+    /// Name of 'address' argument used as override for the address provided by gRPC operation.
+    /// </summary>
+    internal const string AddressArgumentName = "address";
+
+    /// <summary>
+    /// An instance of the HttpClient class.
+    /// </summary>
+    private readonly HttpClient _httpClient;
+
+    /// <summary>
+    /// Creates an instance of a <see cref="GrpcOperationRunner"/> class.
+    /// </summary>
+    /// <param name="httpClient">An instance of the HttpClient class.</param>
+    public GrpcOperationRunner(HttpClient httpClient)
+    {
+        this._httpClient = httpClient;
+    }
+
+    /// <inheritdoc/>
+    public async Task<object> RunAsync(GrpcOperation operation, IDictionary<string, string> arguments, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(operation, $"The {nameof(operation)} parameter is not set to an instance of an object.");
+        Verify.NotNull(arguments, $"The {nameof(arguments)} parameter is not set to an instance of an object.");
+
+        var address = this.GetAddress(operation, arguments);
+
+        var channelOptions = new GrpcChannelOptions { HttpClient = this._httpClient, DisposeHttpClient = false };
+
+        using (var channel = GrpcChannel.ForAddress(address, channelOptions))
+        {
+            var requestType = BuildGrpcOperationDataContractType(operation.Request);
+
+            var responseType = BuildGrpcOperationDataContractType(operation.Response);
+
+            var method = new Method<object, object>
+            (
+                MethodType.Unary,
+                operation.FullServiceName,
+                operation.Name,
+                this.CreateMarshaller<object>(requestType),
+                this.CreateMarshaller<object>(responseType)
+            );
+
+            var invoker = channel.CreateCallInvoker();
+
+            var request = this.GenerateOperationRequest(operation, requestType, arguments);
+
+            return await invoker.AsyncUnaryCall(method, null, new CallOptions(cancellationToken: cancellationToken), request).ConfigureAwait(false);
+        }
+    }
+
+    /// <summary>
+    /// Returns address of a channel that provides connection to a gRPC server.
+    /// </summary>
+    /// <param name="operation">The gRPC operation.</param>
+    /// <param name="arguments">The gRPC operation arguments.</param>
+    /// <returns>The channel address.</returns>
+    private string GetAddress(GrpcOperation operation, IDictionary<string, string> arguments)
+    {
+        var address = default(string);
```

Comment:
Kinda weird... just make it `string? address = null;`?

Reply:
Just stuck to the same way of declaring variables in this PR. Fixed.